### ISSUE TITLE
Unable to run Windows Exe using Absolute Path

### ISSUE THREAD

**Issue by sohang3112** [nushell/nushell#8174]
### Describe the bug

Windows Powershell & CMD both allow running EXE from absolute path. But this doesn't work in Nushell.

### How to reproduce

Try to run any EXE using an absolute path in Windows. For example, this raises an error:

```
$ "C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe"
Error: nu::parser::parse_mismatch (link)

  × Parse mismatch during operation.
   ╭─[entry #10:1:1]
 1 │ "C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe"
   · ────────────────────────────────────────────┬────────────────────────────────────────────
   ·                                             ╰── expected any shape
   ╰────
```

**Note:** I have verified that the EXE path is correct.

### Expected behavior

I expected the EXE to run (whose absolute path I put here).

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.75.0                                   |
| branch             |                                          |
| commit_hash        | 17a265b197ad6bb77db5a07942a9f160ec260eb0 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)      |
| rust_channel       | 1.66.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)      |
| pkg_version        | 0.75.0                                   |
| build_time         | 2023-02-01 08:01:00 -06:00               |
| build_rust_channel | release                                  |
| features           | database, default, trash, which, zip     |
| installed_plugins  |                                          |

### Additional context

Some existing issues have been raised about this, but they were all marked stale and closed without any solution.

**Comment by kubouch** [comment]
That's because `""` string support escapes: the backslashes are interpreted as escapes. Try `''` instead.

**Comment by fdncred** [comment]
I'm hoping at some point we can change the way that our single quotes and double quotes are interpreted. I'd prefer to have something like a sigil that means interpret escapes in the following string like `print @"\e[31mHello\e[0m"` would print a red Hello but `print "\e[31mHello\e[0m"` would just print the string.

**Comment by sohang3112** [comment]
> That's because `""` string support escapes: the backslashes are interpreted as escapes. Try `''` instead.

@kubouch That doesn't work either - the program doesn't run, instead its file path is printed. I tried using both single and double quotes:

```
$ 'C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe'                            
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe
```

```
D:\Work〉"C:/Program Files/Microsoft Visual Studio/2022/Community/MSBuild/Current/Bin/MSBuild.exe"                            C:/Program Files/Microsoft Visual Studio/2022/Community/MSBuild/Current/Bin/MSBuild.exe
```

**Comment by fdncred** [comment]
I think you have to use backticks with paths with spaces in them. If you try to autocomplete, I believe it autocompletes with backticks too.

**Comment by ChrisDenton** [comment]
I have the same issue. Putting a quoted string (using any quote type) on the command line merely echoes that string. The only thing that works is bare (unquoted) paths which doesn't allow spaces. I have to use `^` to run an (external) command.
```
^'C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe'
```

**Comment by fdncred** [comment]
We're looking for suggestions to make this better. 

My suggestion was to use a sigil like `@"some string"` or `@'some string'` to indicate to nushell that escapes should be processed. We're not sure that's the right thing to do. In the case of running an external, I'd like `^"c:\some path\to an external\file.exe"` to work. Or maybe we should just try to run quoted strings in the command position? Not sure.

Please let us know if you have better suggestions.

**Comment by ChrisDenton** [comment]
A sigil sounds reasonable to me, personally. But I guess there's a tension here between doing what other people are used to (which depends on where they're coming from) and being consistent across platforms (so a nushell user can switch between Linux and Windows effortlessly).

**Comment by sohang3112** [comment]
>  I have to use ^ to run an (external) command.

Thanks @ChrisDenton - that worked! (I'm using version 0.75 - that's the latest available on `winget`).

**Comment by bobhy** [comment]
I think changing the escape interpretation in string behavior is the wrong direction for resolving this issue.  

Note that Nu doesn't interpret a quoted string as a call under normal circumstances, why should user expect it to work for windows executables?  e.g:
```rust
〉"ls"
ls

〉ls
╭───┬──────────────────┬──────┬────────┬──────────────╮
│ # │       name       │ type │  size  │   modified   │
├───┼──────────────────┼──────┼────────┼──────────────┤
│ 0 │ df.nu            │ file │ 1.4 KB │ 4 hours ago  │
╰───┴──────────────────┴──────┴────────┴──────────────╯
```

Maybe it would satisfy OP if we came up with syntax to escape a blank in a bare string?

However it's done, though, it can be  rarely used syntax because the problem is rarely encountered (sez me!).  Even on Windows most executables are on the path and invoked by file name and most file names do *not* have blanks.  It's just that pesky "\Program Files*"! 

So, suppose we provide syntax for escaping a blank in a bare string? 
 
Maybe the rule is, you can use `\ ` in a bare string, period, end of story.  Or, if we want to restrict it a bit, say that's only allowed in the first bare string in the command line, or only if it's first in the command line *and* prefixed with `^`.  The '\ ` will look somewhat familar to (experienced) shell users.

**Comment by sohang3112** [comment]
@bobhy IMO it would be convinient to be able to run like this: `^"C:\path\to\app.exe"`. This already works with single quotes, so it shouldn't be too hard to make `^` followed by double quotes behave the same. 
The reason is that when we right click an application in Windows and click *Copy as path*, Windows copies the path with double quotes surrounding it.

**Comment by sohang3112** [comment]
@bobhy As an alternative, Nushell could have a `run` command that simply runs the executable with its given arguments. Since the full executable path (in quotes) would be passed to the `run` command as a normal CLI argument, there won't be any complications.

**Comment by fdncred** [comment]
There already is a `run-external` command but the issue here is double quotes. Double quotes interpret escapes in nushell.

**Comment by eugenesvk** [comment]
> like this: `^"C:\path\to\app.exe"`.

that would still pale in convenient comparison to `"C:\path\app.exe"`
Very frequent actions should require no effort, not Shift+6

**Comment by fdncred** [comment]
I agree with your myself but if we're going to keep our json compatibility and heritage, we have to interpret escapes within double quotes. It's one of the trade-offs that we've accepted but I hate it with a passion because all windows users suffer because of it. It would be nice to somehow have a compromise between json compatibility and escapes on windows.

**Comment by eugenesvk** [comment]
Does json compatibility make it impossible to have a special case of
- if first string resolves to a binary, treat it as literal ^'path\to\'?

But also, even single quotes don't work without ^, and those are even easier to type than. double
(even though that doesn't help with autoescaping mechanisms that copy paths with ")

**Comment by eugenesvk** [comment]
> Note that Nu doesn't interpret a quoted string as a call under normal circumstances, why should user expect it to work for windows executables? e.g

because user expects it to work under normal circumstances outside of Windows. It's weird that the most common operation in a shell - running another program - doesn't follow the path of least resistance and requires prefixes

**Comment by fdncred** [comment]
I don't have issues on Windows running binaries.
```
❯ ~/.cargo/bin/fd
msvs.nu
cheat-sheet.nu
build-all-plugins.nu
nu_scripts\
nu_scripts\typos.toml
nu_scripts\toolkit.nu
...
```
or with \
```
❯ ~\.cargo\bin\fd
msvs.nu
cheat-sheet.nu
build-all-plugins.nu
nu_scripts\
nu_scripts\typos.toml
nu_scripts\toolkit.nu
...
```

**Comment by eugenesvk** [comment]
Now just add a little space to your path and you'll start to have problems...

> if you're using single quotes, you're telling nushell it's a string. you need to tell it's an external with the caret.

I'm telling nu to not treat spaces as separators. Also, when would you ever want to type a string and get it simply echoed back to you?

**Comment by fdncred** [comment]
with spaces in your path you have to use backtick quotes.
